src/main/java/com/programmers/java/Calculator.java

```diff
@@ -0,0 +1,99 @@
+package com.programmers.java;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Stack;
+
+public class Calculator {
+    static int index;
+    static ArrayList<String> history = new ArrayList<>();
+    public static BigDecimal calculate(String expression) throws Exception {
+        index = 0;
+        String postExp = toPostfix(expression);
+        index = 0;
+        BigDecimal answer = postfixCalculator(postExp);
+        history.add(expression+" = "+answer);
+        return answer;
+    }
+
+    public static BigDecimal postfixCalculator(String postExp) throws Exception {
+        Stack<BigDecimal> numberStack = new Stack<>();
+        while (index < postExp.length()) {
+            char ch = postExp.charAt(index);
+            if (Character.isDigit(ch)) {
+                numberStack.push(new BigDecimal(tokenizer(postExp)));
+            } else if (isOperator(ch)) {
+                numberStack.push(performOperation(ch, numberStack.pop(), numberStack.pop()));
+            }
+            index++;
+        }
+        return numberStack.pop();
+    }
+
+    private static String toPostfix(String expression) throws Exception {
+        Stack<Character> operatorStack = new Stack<>();
+        StringBuilder intermediateBuilder = new StringBuilder();
+        while(index < expression.length()) {
+            char ch = expression.charAt(index);
+            if(Character.isDigit(ch)) {
+                intermediateBuilder.append(tokenizer(expression)+ " ");

```

Comment:
StringBuilder를 사용하는 이유는 jvm의 constant pool을 최대한 적게 사용하는데 있습니다. 
(자바 밑바닥 스터디 키워드에 있어요~)
따라서 builder를 사용하려면 append 체이닝을 적극 활용해줍시다!

```suggestion
                intermediateBuilder.append(tokenizer(expression)).append(" ");
```